Ctest for the attestation failed

When I build with LVI mitigation, run:
```
cmake .. \
-DLVI_MITIGATION=ControlFlow \
-DLVI_MITIGATION_BINDIR=/usr/local/lvi-mitigation/bin
make
```
I got the result:
```
[ 15%] Linking CXX executable safecrt_enc-lvi-cfg
/usr/bin/ld: cannot find -link-lvi-mitigation
/usr/bin/ld: CMakeFiles/safecrt_enc-lvi-cfg.dir/__/common/test.cpp.o: unable to initialize decompress status for section .debug_str
/usr/bin/ld: CMakeFiles/safecrt_enc-lvi-cfg.dir/__/common/test.cpp.o: unable to initialize decompress status for section .debug_str
CMakeFiles/safecrt_enc-lvi-cfg.dir/__/common/test.cpp.o: file not recognized: File format not recognized
clang: error: linker command failed with exit code 1 (use -v to see invocation)
tests/safecrt/enc/CMakeFiles/safecrt_enc-lvi-cfg.dir/build.make:149: recipe for target 'tests/safecrt/enc/safecrt_enc-lvi-cfg' failed
make[2]: *** [tests/safecrt/enc/safecrt_enc-lvi-cfg] Error 1
CMakeFiles/Makefile2:8668: recipe for target 'tests/safecrt/enc/CMakeFiles/safecrt_enc-lvi-cfg.dir/all' failed
make[1]: *** [tests/safecrt/enc/CMakeFiles/safecrt_enc-lvi-cfg.dir/all] Error 2
Makefile:181: recipe for target 'all' failed
make: *** [all] Error 2
```
It works on Ubuntu18.04, SGX1+FLC model
My environment path is:
```
/home/dylan/.local/bin:/home/dylan/bin:.:/usr/local/jdk1.8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```
I have installed the  dependencies according to the guidance document, contents under the director `/usr/local/lvi-mitigation/bin` are:
```
drwxr-xr-x 2 root root  4096 Mar 15 14:51 ./
drwxr-xr-x 4 root root  4096 Mar 14 11:38 ../
lrwxrwxrwx 1 root root    57 Mar 14 11:38 as -> /usr/local/lvi-mitigation/external/toolset/ubuntu18.04/as*
-rwxr-xr-x 1 root root   231 Mar 15 19:49 clang-8*
-rwxr-xr-x 1 root root   233 Mar 15 19:49 clang++-8*
lrwxrwxrwx 1 root root    16 Mar 14 11:38 clang-8_symlink -> /usr/bin/clang-8*
lrwxrwxrwx 1 root root    18 Mar 14 11:38 clang++-8_symlink -> /usr/bin/clang++-8*
-rwxr-xr-x 1 root root   227 Mar 15 19:49 g++*
-rwxr-xr-x 1 root root   227 Mar 15 19:49 gcc*
lrwxrwxrwx 1 root root    12 Mar 14 11:38 gcc_symlink -> /usr/bin/gcc*
-r-xr-xr-x 1 root root   660 Mar 14 11:38 generate_wrapper*
lrwxrwxrwx 1 root root    12 Mar 14 11:39 g++_symlink -> /usr/bin/g++*
-r-xr-xr-x 1 root root  2649 Mar 14 11:38 install_lvi_mitigation_bindir*
-r-xr-xr-x 1 root root  1093 Mar 14 11:38 invoke_compiler*
lrwxrwxrwx 1 root root    57 Mar 14 11:38 ld -> /usr/local/lvi-mitigation/external/toolset/ubuntu18.04/ld*
lrwxrwxrwx 1 root root    62 Mar 15 14:50 ld.gold -> /usr/local/lvi-mitigation/external/toolset/ubuntu18.04/ld.gold*
-r-xr-xr-x 1 root root 11514 Mar 14 11:38 lvi-mitigation.py*
lrwxrwxrwx 1 root root    62 Mar 15 14:51 objdump -> /usr/local/lvi-mitigation/external/toolset/ubuntu18.04/objdump*
```
There is another error when I first run `make`
```
/usr/bin/as: unrecognized option '-mlfence-before-indirect-branch=register'
```
Then I create a soft link for `as` in the directory `/usr/bin` like, it works.
```
dylan@dylan:/usr/bin$ ll | grep as
lrwxrwxrwx  1 root  root          57 Mar 15 17:28 as -> /usr/local/lvi-mitigation/external/toolset/ubuntu18.04/as*
```
But  after `make` again, I got the result showed at the beginning: `[ 15%] Linking CXX executable safecrt_enc-lvi-cfg
/usr/bin/ld: cannot find -link-lvi-mitigation`

@qiucwang Here is the error information when running the attestation sample.

```
dylan@dylan:/opt/openenclave/share/openenclave/samples/attestation/build$ sudo make runsgxremote OE_LOG_LEVEL=INFO
[ 18%] Built target attestation_host
[ 22%] Built target public_key_a
[ 50%] Built target common
[ 63%] Built target enclave_b
[ 72%] Built target enclave_b_signed
[ 77%] Built target public_key_b
[ 90%] Built target enclave_a
[100%] Built target enclave_a_signed
[100%] Built target sign
Host: Creating two enclaves
Host: Enclave library /opt/openenclave/share/openenclave/samples/attestation/build/enclave_a/enclave_a.signed
2021-03-24T10:37:54+1300.195722Z [(H)INFO] tid(0x7fd660f32740) | Processor supports AVX instructions [/source/openenclave/host/sgx/linux/xstate.c:_is_xgetbv_supported:33]
2021-03-24T10:37:54+1300.195840Z [(H)INFO] tid(0x7fd660f32740) | Processor supports AVX instructions [/source/openenclave/host/sgx/linux/xstate.c:_is_xgetbv_supported:33]
2021-03-24T10:37:54+1300.195855Z [(H)INFO] tid(0x7fd660f32740) | Value of XFRM to be set in enclave is 7
 [/source/openenclave/host/sgx/sgxload.c:_detect_xfrm:121]
2021-03-24T10:37:54+1300.204870Z [(H)INFO] tid(0x7fd660f32740) | Loading libsgx_enclave_common.so.1
 [/source/openenclave/host/sgx/sgx_enclave_common_wrapper.c:_load_sgx_enclave_common_impl:110]
2021-03-24T10:37:54+1300.205154Z [(H)INFO] tid(0x7fd660f32740) | Loaded libsgx_enclave_common.so.1
 [/source/openenclave/host/sgx/sgx_enclave_common_wrapper.c:_load_sgx_enclave_common_impl:126]
Enclave1: ***/opt/openenclave/share/openenclave/samples/attestation/common/crypto.cpp(80): mbedtls initialized.
2021-03-24T10:37:54+1300.361440Z [(H)WARN] tid(0x7fd660f32740) | In-enclave logging is not supported. To enable, please add 

from "openenclave/edl/logging.edl" import *;

in the edl file.
 [/source/openenclave/host/sgx/create.c:oe_create_enclave:1148]
Host: Enclave successfully created.
Host: Enclave library /opt/openenclave/share/openenclave/samples/attestation/build/enclave_b/enclave_b.signed
2021-03-24T10:37:54+1300.361451Z [(H)INFO] tid(0x7fd660f32740) | Processor supports AVX instructions [/source/openenclave/host/sgx/linux/xstate.c:_is_xgetbv_supported:33]
2021-03-24T10:37:54+1300.361453Z [(H)INFO] tid(0x7fd660f32740) | Value of XFRM to be set in enclave is 7
 [/source/openenclave/host/sgx/sgxload.c:_detect_xfrm:121]
Enclave2: ***/opt/openenclave/share/openenclave/samples/attestation/common/crypto.cpp(80): mbedtls initialized.
2021-03-24T10:37:54+1300.541469Z [(H)WARN] tid(0x7fd660f32740) | In-enclave logging is not supported. To enable, please add 

from "openenclave/edl/logging.edl" import *;

in the edl file.
 [/source/openenclave/host/sgx/create.c:oe_create_enclave:1148]
Host: Enclave successfully created.


Host: ********** Attest enclave_a to enclave_b **********

Host: Requesting enclave_b format settings
Enclave2: ***/opt/openenclave/share/openenclave/samples/attestation/common/dispatcher.cpp(80): get_enclave_format_settings
Host: Requesting enclave_a to generate a targeted evidence with an encryption key
Enclave1: ***/opt/openenclave/share/openenclave/samples/attestation/common/dispatcher.cpp(139): get_evidence_with_public_key
2021-03-24T10:37:54+1300.541699Z [(H)INFO] tid(0x7fd660f32740) | Loading libsgx_dcap_ql.so
 [/source/openenclave/host/sgx/sgxquote.c:_load_sgx_dcap_ql_impl:176]
2021-03-24T10:37:54+1300.541915Z [(H)INFO] tid(0x7fd660f32740) | Loaded libsgx_dcap_ql.so
 [/source/openenclave/host/sgx/sgxquote.c:_load_sgx_dcap_ql_impl:194]
2021-03-24T10:37:54+1300.541920Z [(H)INFO] tid(0x7fd660f32740) | DCAP installed and set for in-process quoting. [/source/openenclave/host/sgx/sgxquote.c:_load_quote_ex_library_once:267]
2021-03-24T10:37:54+1300.541927Z [(H)INFO] tid(0x7fd660f32740) | DCAP only supports ECDSA_P256
 [/source/openenclave/host/sgx/sgxquote.c:oe_sgx_get_supported_attester_format_ids:829]
Enclave1: ***/opt/openenclave/share/openenclave/samples/attestation/common/attestation.cpp(94): oe_serialize_custom_claims
Enclave1: ***/opt/openenclave/share/openenclave/samples/attestation/common/attestation.cpp(105): serialized custom claims buffer size: 121
2021-03-24T10:37:54+1300.541992Z [(H)INFO] tid(0x7fd660f32740) | oe_load_quote_provider libdcap_quoteprov.so
 [/source/openenclave/host/sgx/linux/sgxquoteproviderloader.c:oe_load_quote_provider:25]
2021-03-24T10:37:54+1300.545470Z [(H)WARN] tid(0x7fd660f32740) | sgxquoteprovider: sgx_ql_set_logging_function not found
 [/source/openenclave/host/sgx/linux/sgxquoteproviderloader.c:oe_load_quote_provider:46]
2021-03-24T10:37:54+1300.545477Z [(H)INFO] tid(0x7fd660f32740) | sgxquoteprovider: provider.get_sgx_quote_verification_collateral = 0x7fd660ee87c0
 [/source/openenclave/host/sgx/linux/sgxquoteproviderloader.c:oe_load_quote_provider:59]
2021-03-24T10:37:54+1300.545479Z [(H)INFO] tid(0x7fd660f32740) | sgxquoteprovider: provider.get_sgx_quote_verification_collateral = 0x7fd660ee8700
 [/source/openenclave/host/sgx/linux/sgxquoteproviderloader.c:oe_load_quote_provider:64]
2021-03-24T10:37:54+1300.685099Z [(H)INFO] tid(0x7fd660f32740) | quote_size=4578 [/source/openenclave/host/sgx/sgxquote.c:oe_sgx_qe_get_quote:755]
Enclave1: ***/opt/openenclave/share/openenclave/samples/attestation/common/attestation.cpp(126): generate_attestation_evidence succeeded.
Enclave1: ***/opt/openenclave/share/openenclave/samples/attestation/common/dispatcher.cpp(190): get_evidence_with_public_key succeeded
Host: enclave_a's  public key: 
-----BEGIN PUBLIC KEY-----
MIIBIj  omit....
-----END PUBLIC KEY-----

Host: verify_evidence_and_set_public_key in enclave_b
2021-03-24T10:37:54+1300.701707Z [(H)INFO] tid(0x7fd660f32740) | Calling oe_get_sgx_quote_verification_collateral
 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:63]
2021-03-24T10:37:54+1300.745758Z [(H)INFO] tid(0x7fd660f32740) | pck_crl_issuer_chain_size = 1909
 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:172]
2021-03-24T10:37:54+1300.745769Z [(H)INFO] tid(0x7fd660f32740) | pck_crl_issuer_chain json = 
-----BEGIN CERTIFICATE-----
MIIC omit...
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIIC  omit....
-----END CERTIFICATE-----

 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:174]
2021-03-24T10:37:54+1300.745780Z [(H)INFO] tid(0x7fd660f32740) | root_ca_crl_size = 448
 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:189]
2021-03-24T10:37:54+1300.745782Z [(H)INFO] tid(0x7fd660f32740) | pck_crl_size = 458
 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:204]
2021-03-24T10:37:54+1300.745784Z [(H)INFO] tid(0x7fd660f32740) | pck_crl_size = 1893
 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:220]
2021-03-24T10:37:54+1300.745787Z [(H)INFO] tid(0x7fd660f32740) | tcb_info_size = 4844
 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:235]
2021-03-24T10:37:54+1300.745789Z [(H)INFO] tid(0x7fd660f32740) | qe_identity_issuer_chain_size = 1893
 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:254]
2021-03-24T10:37:54+1300.745790Z [(H)INFO] tid(0x7fd660f32740) | qe_identity_size = 856
 [/source/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:269]
2021-03-24T10:37:54+1300.745893Z [(H)INFO] tid(0x7fd660f32740) | DCAP only supports ECDSA_P256
 [/source/openenclave/host/sgx/sgxquote.c:oe_sgx_get_supported_attester_format_ids:829]
Enclave2: ***/opt/openenclave/share/openenclave/samples/attestation/common/attestation.cpp(201): oe_verify_evidence failed (OE_TCB_LEVEL_INVALID).

Enclave2: ***/opt/openenclave/share/openenclave/samples/attestation/common/dispatcher.cpp(224): verify_evidence_and_set_public_key failed.
Host: verify_evidence_and_set_public_key failed. OE_OK
Host: attestation failed with 1
Host: Terminating enclaves
Enclave1: ***/opt/openenclave/share/openenclave/samples/attestation/common/crypto.cpp(94): mbedtls cleaned up.
Host: Enclave successfully terminated.
Enclave2: ***/opt/openenclave/share/openenclave/samples/attestation/common/crypto.cpp(94): mbedtls cleaned up.
Host: Enclave successfully terminated.
Host:  failed 
2021-03-24T10:37:54+1300.803104Z [(H)INFO] tid(0x7fd660f32740) | _unload_quote_provider libdcap_quoteprov.so
 [/source/openenclave/host/sgx/linux/sgxquoteproviderloader.c:_unload_quote_provider:13]
CMakeFiles/runsgxremote.dir/build.make:57: recipe for target 'CMakeFiles/runsgxremote' failed
make[3]: *** [CMakeFiles/runsgxremote] Error 1
CMakeFiles/Makefile2:73: recipe for target 'CMakeFiles/runsgxremote.dir/all' failed
make[2]: *** [CMakeFiles/runsgxremote.dir/all] Error 2
CMakeFiles/Makefile2:80: recipe for target 'CMakeFiles/runsgxremote.dir/rule' failed
make[1]: *** [CMakeFiles/runsgxremote.dir/rule] Error 2
Makefile:118: recipe for target 'runsgxremote' failed
make: *** [runsgxremote] Error 2
```

--- a/host/sgx/sgxquoteprovider.c
+++ b/host/sgx/sgxquoteprovider.c
@@ -91,12 +91,12 @@ oe_result_t oe_get_sgx_quote_verification_collateral(
         OE_RAISE(OE_QUOTE_PROVIDER_CALL_ERROR);
     }
 
-    if (collateral->version != SGX_QL_QVE_COLLATERAL_VERSION)
+    if (collateral->version != SGX_QL_QVE_COLLATERAL_VERSION_1 &&
+        collateral->version != SGX_QL_QVE_COLLATERAL_VERSION_3)
     {
         OE_RAISE_MSG(
             OE_INVALID_ENDORSEMENT,
-            "Expected version to be %d, but got %d",
-            SGX_QL_QVE_COLLATERAL_VERSION,
+            "Invalid collateral version %d",
             collateral->version);
     }
 

--- a/host/sgx/sgxquoteprovider.h
+++ b/host/sgx/sgxquoteprovider.h
@@ -31,8 +31,14 @@ typedef struct _oe_sgx_quote_provider
 #define SGX_QL_FREE_QUOTE_VERIFICATION_COLLATERAL_NAME \
     "sgx_ql_free_quote_verification_collateral"
 
-/*! Version of the supported SGX quote verification collateral  */
-#define SGX_QL_QVE_COLLATERAL_VERSION (1)
+/**
+ * Version of the supported SGX quote verification collateral, which will
+ * reflect the version of the PCCS/PCS API used to retrieve the collateral.
+ * For V1 and V2 APIs, the ‘version’ field with have a value of 1.
+ * For V3 APIs, the ‘version’ field will have the value of 3."
+ */
+#define SGX_QL_QVE_COLLATERAL_VERSION_1 (1)
+#define SGX_QL_QVE_COLLATERAL_VERSION_3 (3)
 
 OE_EXTERNC_END
 

